nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-properties-loader/src/test/groovy/org/apache/nifi/properties/sensitive/aws/kms/AWSKMSSensitivePropertyProviderIT.groovy

```diff
@@ -0,0 +1,253 @@
+    private static final Logger logger = LoggerFactory.getLogger(AWSKMSSensitivePropertyProviderIT.class)
+    protected final static String CREDENTIALS_FILE = System.getProperty("user.home") + "/aws-credentials.properties";
+    private static String[] knownGoodKeys = []
+    private static AWSKMSClient client
+
+    /**
+     * This method creates a CMK, DEK, and an alias to that DEK for exercising the AWS KMS calls.
+     *
+     * @throws Exception
+     */
+    @BeforeClass
+    static void setUpOnce() throws Exception {
+        final FileInputStream fis
+        try {
+            fis = new FileInputStream(CREDENTIALS_FILE)
+        } catch (FileNotFoundException e1) {
+            fail("Could not open credentials file " + CREDENTIALS_FILE + ": " + e1.getLocalizedMessage());
+            return
+        }
+        final PropertiesCredentials credentials = new PropertiesCredentials(fis)
+
+        // We're copying the properties directly so the standard builder works.
+        System.setProperty("aws.accessKeyId", credentials.AWSAccessKeyId)
+        System.setProperty("aws.secretKey", credentials.AWSSecretKey)
+        System.setProperty("aws.region", "us-east-2")
+
+        client = AWSKMSClientBuilder.standard().build() as AWSKMSClient
+
+        // generate a cmk
+        CreateKeyRequest cmkRequest = new CreateKeyRequest().withDescription("CMK for unit tests")
+        CreateKeyResult cmkResult = client.createKey(cmkRequest)
+
+        // from the cmk, generate a dek
+        GenerateDataKeyRequest dekRequest = new GenerateDataKeyRequest().withKeyId(cmkResult.keyMetadata.getKeyId()).withKeySpec("AES_128")
+        GenerateDataKeyResult dekResult = client.generateDataKey(dekRequest)
+
+        // add an alias to the dek
+        final String aliasName = "alias/aws-kms-spp-integration-test-" + UUID.randomUUID().toString()
+        CreateAliasRequest aliasReq = new CreateAliasRequest().withAliasName(aliasName).withTargetKeyId(dekResult.getKeyId())
+        client.createAlias(aliasReq)
+
+        // re-read the dek so we have the arn
+        DescribeKeyRequest descRequest = new DescribeKeyRequest().withKeyId(dekResult.getKeyId())
+        DescribeKeyResult descResult = client.describeKey(descRequest)
+
+        knownGoodKeys = [
+                dekResult.getKeyId(),
+                descResult.keyMetadata.getArn(),
+                aliasName
+        ]
+    }
+
+    @Before
+    void setUp() throws Exception {
+    }
+
+    @After
+    void tearDown() throws Exception {
+    }
+
+    /**
+     * This method schedules the deletion of the CMK created during setup.  The delete will cascade to the DEK and DEK alias.
+     */
+    @AfterClass
+    static void tearDownOnce() {
+        if (knownGoodKeys.size() > 0) {
+            ScheduleKeyDeletionRequest req = new ScheduleKeyDeletionRequest().withKeyId(knownGoodKeys[0]).withPendingWindowInDays(7)
+            client.scheduleKeyDeletion(req)
+        }
+    }
```

Comment:
Would be good to delete the system properties containing the AWS credentials here, with a flag that indicated if they were overwritten during `#setUpOnce()`.

Reply:
Good idea.  I've added logic to save/restore those system properties before/after the test runs.